mri/reconstructors/base.py

```diff
@@ -102,14 +102,10 @@ def initialize_gradient_op(self, **extra_args):
             verbose=self.verbose,
             **extra_args,
         )
-        self.cost_op = GenericCost(
-            gradient_op=self.gradient_op,
-            prox_op=self.prox_op,
-            verbose=self.verbose >= 20,
-        )
 
     def reconstruct(self, kspace_data, optimization_alg='pogm',
-                    x_init=None, num_iterations=100, **kwargs):
+                    x_init=None, num_iterations=100, cost_op_kwargs={},
```

Comment:
generally we should avoid mutable as defaults, so you could have `None` here and test `if cost_op_kwargs is None: cost_op_kwargs = {}`